library/std/src/f64/soft/tests.rs

```diff
@@ -0,0 +1,49 @@
+use crate::f64::soft::{Positive, PositiveFinite, Representation, Sign, pow2};
+
+#[test]
+fn test_representation() {
+    assert_eq!(Representation::new(-1.5f64), Representation {
+        sign: Sign::Negative,
+        abs: Positive::Finite(PositiveFinite { exp: -52, mantissa: 3 << 51 }),
+    });
+
+    assert_eq!(Representation::new(f64::MIN_POSITIVE), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: -1074, mantissa: 1 << 52 }),
```

Comment:
...this exponent is a fair bit away from `-1021`, so I am going to have to go over  a number of things very carefully, and I am not immediately seeing an explanation of the encoding choice in this module. I'd like to see a comment somewhere (perhaps on the struct, or on the associated constants, or this test?), that explains why what I am seeing here is Actually Correct.